bentoml/_internal/bento/bento.py

```diff
@@ -170,45 +183,45 @@ def save(
         with bento_store.register(self.tag) as bento_path:
             # TODO: handle non-OSFS store types
             os.rmdir(bento_path)
-            os.rename(self.fs.getsyspath("/"), bento_path)
-            self.fs.close()
+            os.rename(self._fs.getsyspath("/"), bento_path)
+            self._fs.close()
 
         return bento_store.get(self.tag)
 
     def export(self, path: str):
         mirror(
-            self.fs, fs.open_fs(path, create=True, writeable=True), copy_if_newer=False
+            self._fs, fs.open_fs(path, create=True, writeable=True), copy_if_newer=False
         )
 
     def push(self):
         pass
 
     def validate(self):
-        return self.fs.isfile(BENTO_YAML_FILENAME)
+        return self._fs.isfile(BENTO_YAML_FILENAME)
 
 
 class BentoStore(Store[Bento]):
     def __init__(self, base_path: PathType):
         super().__init__(base_path, Bento)
 
 
-class BentoMetadata(UserDict):
-    def dump(self, stream: t.IO):
+class BentoMetadata(UserDict[str, t.Any]):
+    def dump(self, stream: t.IO[str]):
         return yaml.dump(self.data, stream, sort_keys=False)
 
     @classmethod
     def from_build_args(
         cls, tag: Tag, svc: "Service", labels: t.Dict[str, t.Any], models: t.List[str]
     ):
         bento_metadata = cls()
-        bento_metadata["service"] = svc._import_str
+        bento_metadata["service"] = svc._import_str  # type: ignore[reportPrivateUsage]
```

Comment:
is it possible to do this ignore for everything in  the `Service` class?

Reply:
I think the better way to do this is to have a property to access the protected attributes.
```python
@property
def import_str(self):
	return self._import_str
```

Reply:
> is it possible to do this ignore for everything in  the `Service` class? 

Not that I know of; I'll take a look.

Reply:
> I think the better way to do this is to have a property to access the protected attributes.
> 
> ```python
> @property
> def import_str(self):
> 	return self._import_str
> ```

Hm I thought I replied to this, but the reason I didn't do this was because that would expose these to the user, and I don't know that that's what we want.

Reply:
that's right, this is not supposed to be exposed to the user and only meant to be used within BentoML, same for `_apis` and `_runners`.

Reply:
would a function like
```python
def privateattr(cls_, attr: str):
	return getattr(cls, f"_{attr}")
```
fix our issue?
I'm fine with writing the `type: ignore` but it will make our codebase very tiresome to look into :(

Reply:
We should probably just have a service subclass that we can cast into when we use them internally that surfaces the attributes.

Reply:
I'd say let's draft this in another PR, this is large enough as it is.

Reply:
noted